packages/optivum-scrapper/src/utils.ts

```diff
@@ -80,88 +81,78 @@ export const parseClassCode = (code: string) => {
-            roomId = parseUnitLink(roomTag)?.id ?? null;
-            roomCode = roomTag.textContent ?? null;
+            room = { id: parseUnitLink(roomTag)?.id ?? null, short: roomTag.textContent! };
         }
         roomTag.remove();
     }
 
+    // Teacher
+    let teacher: { id: string | null; short: string } | null = null;
     const teacherTag = fragment.querySelector('.n');
     if (teacherTag !== null) {
-        teacherId = parseUnitLink(teacherTag)?.id ?? null;
-        teacherInitials = teacherTag.textContent;
+        teacher = { id: parseUnitLink(teacherTag)?.id ?? null, short: teacherTag.textContent! };
         teacherTag.remove();
     }
-    const classes = splitByComma(fragment).map((classDocument): Class => {
-        const classTag = classDocument.querySelector('.o');
-        const id = parseUnitLink(classTag)?.id ?? null;
-        const code = classTag?.textContent ?? null;
-        classTag?.remove();
-        let groupCode = commonGroupCode ?? classDocument.textContent?.trim().replace('-', '') ?? null;
-        if (groupCode === '') groupCode = null;
-        return {
-            id,
-            code,
-            groupCode,
-        };
-    });
+
+    // Classes and common groups
+    let classes: LessonClass[] = [];
+    if (fragment.querySelector('.o')) {
+        classes = splitByComma(fragment).map((classDocument): LessonClass => {
+            const classTag = classDocument.querySelector('.o')!;
+            const id = parseUnitLink(classTag)?.id ?? null;
+            const short = classTag.textContent ?? null;
+            classTag.remove();
+            let groupShort: string | null = groupShort_ ?? classDocument.textContent?.trim().replace('-', '') ?? null;
+            if (groupShort === '') groupShort = null;
+            return {
+                id,
+                short,
+                groupShort,
+            };
+        });
+    } else if (groupShort_ !== null) {
+        classes.push({ id: null, short: null, groupShort: groupShort_ });
+    }
     return {
-        subjectCode,
-        teacherId,
-        teacherInitials,
-        roomId,
-        roomCode,
+        subjectId,
+        teacher,
+        room,
         classes,
         comment: null,
-        interclassGroupCode,
+        interclassGroupId,
     };
 };
 
-export function getTeacherKey(currentUnit: { id: number; symbol: string }, id: number | null, code: string | null) {
-    if (currentUnit.symbol === 'n') return currentUnit.id.toString();
-    return id !== null || code !== null ? id?.toString() ?? `#${code}` : null;
-}
-
-export function getRoomKey(currentUnit: { id: number; symbol: string }, id: number | null, code: string | null) {
-    if (currentUnit.symbol === 's') return currentUnit.id.toString();
-    return id !== null || code !== null ? id?.toString() ?? `#${code}` : null;
-}
+export const getUnitKey = (unit: { id: string | null; short: string } | null) => (unit ? unit.id ?? `@${unit.short}` : null);
```

Comment:
```suggestion
export const getUnitKey = (unit: { id: string | null; short: string }) => unit.id ?? `@${unit.short}`;
```

Ta funkcja nie powinna przyjmować `null`, null-check powinien być wykonywany przed użyciem tej funkcji.

Ten kod nie będzie potrzebował non-null asserta:
https://github.com/dominik-korsa/timetable-api/pull/33/files#diff-e748f342605d3ef97d0c49c9298db023a86ed12d5ffe5484428f434fa1ccadb0R154